src/main/java/hudson/plugins/sshslaves/JavaProvider.java

```diff
@@ -62,4 +71,31 @@
         return ExtensionList.lookup(JavaProvider.class);
     }
 
+    /**
+     * Gets minimal required Java version.
+     * @since TODO
+     */
+    @Nonnull
+    public static VersionNumber getMinJavaLevel() {
+        // Try the new API first, since Jenkins TODO (JENKINS-43500)
+        // TODO: Get rid of it once Jenkins core requirement is bumped
+        try {
+            Method method = Jenkins.class.getMethod("getJavaMinLevel");
+            Object res = method.invoke(null);
+            if (res instanceof VersionNumber) {
+                return (VersionNumber) res;
+            }
+        } catch(SecurityException | NoSuchMethodException | IllegalAccessException | InvocationTargetException ex) {
+            // Fallback to the default behavior, not supported yet
+        }
+        
+        // Now use the known map of the previous updates
+        final VersionNumber version = Jenkins.getVersion();
+        if (version == null) {
+            // Version cannot be determined, assume it's an old one to retain compatibility.
+            return JAVA_LEVEL_7;
```

Comment:
I assume this is JAVA_LEVEL_7 rather than JAVA_LEVEL_6 because the plugin is based on Jenkins 1.625 (first LTS to mandate Java 7) and we've had one or more ugly bug reports (in the git plugin, anyway) from users trying to mix Java 7 master with Java 6 agents.